tests/Doctrine/Tests/Common/Persistence/Mapping/ClassMetadataFactoryTest.php

```diff
@@ -34,6 +35,17 @@ public function testGetCacheDriver()
         self::assertSame($cache, $this->cmf->getCacheDriver());
     }
 
+    public function testGetMetadataForAnonymousClass() : void
+    {
+        $object = new class () extends stdClass {
+        };
+
+        $this->expectException(MappingException::class);
+        $this->expectExceptionMessage('anonymous');
```

Comment:
If we drop this line, we can revert all the changes made in `src`, the test will still pass

Reply:
Oh right that must be because the bug is specific to Windows… as pointed out in https://github.com/doctrine/persistence/pull/126#issuecomment-694439226

Reply:
No, it's wrong, because PHP generate custom class name for anonymous classes and we must check, what this exception throws by anonymous class. I understand problem, but in this case we must use one of those cases: generate expected message per PHP version or collect all messages and find universal message. For more details, please see #125 (we close this PR, because this problem must be merged to 1.3 and share to next versions).